sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsDeviceMemoryStore.scala

```diff
@@ -264,17 +256,20 @@ class RapidsDeviceMemoryStore(
       table.close()
     }
 
-    override def meta: TableMeta = {
-      chunkedPacker.getMeta
+    private lazy val (cachedMeta, cachedPackedSize) = {
+      withResource(makeChunkedPacker) { cp =>
+        (cp.getMeta, cp.getTotalContiguousSize)
```

Comment:
I'm not sure I understand what is happening here.  Doesn't this close the ChunkedPacker?  Is the metadata and total contiguous size all we need here?
Previously we held the chunked packer open until `free()`.

Reply:
I know @revans2 had issues where the chunked packer was closed because of a failed host allocation, and a re-attempt to spill to disk. I'll review the changes around here also today.

Reply:
Ok yes this is a way to get around the fact that the chunked packer, and its metadata, are tied together. And because a failed host alloc in the past would have closed the instance chunked packer in the `RapidsTable` (https://github.com/NVIDIA/spark-rapids/pull/9656/files#diff-8816e7aa7f45c4fd7b0e557d63f30d3b3538e4b71633eb7ffa8d98713bb58dc7L122) via the copy iterator.

With the changes here, we are creating the packer only when we need it => at copy time, and to get the metadata.

We should clean this up in cuDF so we can get the metadata and contiguous size without instantiating a packer.